perfkitbenchmarker/vm_util.py

```diff
@@ -512,6 +546,28 @@ def GenerateSSHConfig(vms):
                'ssh -F {0} <vm name>'.format(target_file))
 
 
+def RunningOnWindows():
+  """Returns True if PKB is running on Windows."""
+  return os.name == WINDOWS
+
+
+def ExecutableOnPath(executable_name):
+  """Return True if the given executable can be found on the path."""
+  cmd = ['where'] if RunningOnWindows() else ['which']
+  cmd.append(executable_name)
+
+  shell_value = RunningOnWindows()
+  process = subprocess.Popen(cmd,
+                             shell=shell_value,
+                             stdout=subprocess.PIPE,
+                             stderr=subprocess.PIPE)
+  process.wait()
```

Comment:
Unlikely given the commands used, but this [risks deadlock](https://docs.python.org/2/library/subprocess.html#subprocess.Popen.wait) if `cmd` produces a lot of output. Prefer check_output, communicate, or opening a handle to `os.devnull` and using it for stdout and stderr.